Code/MinimalLib/common.h

```diff
@@ -139,15 +139,16 @@ RWMol *mol_from_input(const std::string &input,
     res = nullptr;
   }
   if (res) {
+    unsigned int failedOp;
+    unsigned int sanitizeOps = MolOps::SANITIZE_ADJUSTHS;
     try {
       if (sanitize) {
-        unsigned int failedOp;
-        unsigned int sanitizeOps = MolOps::SANITIZE_ALL;
+        sanitizeOps = MolOps::SANITIZE_ALL;
         if (!kekulize) {
           sanitizeOps ^= MolOps::SANITIZE_KEKULIZE;
         }
-        MolOps::sanitizeMol(*res, failedOp, sanitizeOps);
       }
+      MolOps::sanitizeMol(*res, failedOp, sanitizeOps);
```

Comment:
Since all you want in the end is to be sure that there are implicit Hs, I think this is clearer and more explicit as:
```
      if (sanitize) {
        unsigned int failedOp;
        unsigned int sanitizeOps = MolOps::SANITIZE_ALL;
        if (!kekulize) {
          sanitizeOps ^= MolOps::SANITIZE_KEKULIZE;
        }
        MolOps::sanitizeMol(*res, failedOp, sanitizeOps);
      } else {
        res->updatePropertyCache(false);
      }
```
You don't actually want `SANITIZE_ADJUSTHS`, which does additional work that the user may not want done (they did, after all, specify that sanitization shouldn't happen).

Reply:
@greglandrum I do agree that's cleaner - I also wasn't entirely happy with my solution. Done.